### PR TITLE
(fix) Next-Auth integration

### PR DESCRIPTION
**Pull request by siddhsuresh**
Closes: https://github.com/blitz-js/blitz/issues/4143

### What are the changes and their implications?

- Change the dynamics imports to regular imports.
- Fix the webpack alias configuration
- Add a postinstall patch for good measure.
- Update pnpm in the workflow to v8.5.1

## Bug Checklist

- [x] Changeset added (run `pnpm changeset` in the root directory)

### CODE REVIEW

File: .changeset/small-monkeys-battle.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+"toolkit-app": patch
```
Comment (siddhsuresh) [inline .changeset/small-monkeys-battle.md:2]:
```suggestion
```

File: packages/blitz-auth/scripts/postinstall.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    }
+
+    root = path.join(blitzPkgLocation, "../")
+  }
+  return path.join(root, ".blitz")
+}
+
+async function patchNextAuth() {
+  try {
+    console.log("1")
```
Comment (flybayer) [inline packages/blitz-auth/scripts/postinstall.js:39]: 🧹

File: packages/blitz-auth/scripts/postinstall.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+      "..",
+      "package.json",
+    ))
+    nextAuthPackageJsonContents.exports = {
+      ...nextAuthPackageJsonContents.exports,
+      "./core/*": "./core/*.js",
+    }
+    console.log(nextAuthPackageJsonContents)
+    fs.writeFileSync(nextAuthPackageJson, JSON.stringify(nextAuthPackageJsonContents, null, 2))
+    console.log("Patched next-auth package.json exports")
```
Comment (flybayer) [inline packages/blitz-auth/scripts/postinstall.js:57]: move these console logs to debug logs

File: packages/blitz-auth/src/server/adapters/next-auth/webpack.ts [hunk 4]
```diff
@@ -1,14 +1,19 @@
 //@ts-nocheck
+import path from "path"
+
 export function withNextAuthAdapter(nextConfig) {
```
Comment (flybayer) [inline packages/blitz-auth/src/server/adapters/next-auth/webpack.ts:4]: needs fixed to retain user webpack config

File: apps/toolkit-app/.env [hunk 5]
```diff
@@ -1,3 +1,6 @@
 # This env file should be checked into source control
 # This is the place for default values for all environments
 # Values in `.env.local` and `.env.production` will override these values
+
+GITHUB_CLIENT_ID= dcd02983e7e911bd169f
+GITHUB_CLIENT_SECRET = 5c3b2bac7d9bc55ba9f99d0d89a8ab4e655ff5bc
```
Comment (siddhsuresh) [inline apps/toolkit-app/.env:6]:
```suggestion
```

File: packages/blitz-auth/src/server/auth-sessions.ts [hunk 6]
```diff
@@ -189,7 +189,7 @@ export async function getBlitzContext(): Promise<Ctx> {
   const req = new IncomingMessage(new Socket()) as IncomingMessage & {
     cookies: {[key: string]: string}
   }
-  req.headers = Object.fromEntries(headers())
+  req.headers = Object.fromEntries(headers() as any)
```
Comment (siddhsuresh) [inline packages/blitz-auth/src/server/auth-sessions.ts:192]:
```suggestion
  // @ts-expect-error TODO
  req.headers = Object.fromEntries(headers())
```

File: packages/blitz-auth/src/server/auth-sessions.ts [hunk 7]
```diff
@@ -189,6 +189,7 @@ export async function getBlitzContext(): Promise<Ctx> {
   const req = new IncomingMessage(new Socket()) as IncomingMessage & {
     cookies: {[key: string]: string}
   }
+  // @ts-expect-error TODO
```
Comment (siddhsuresh) [inline packages/blitz-auth/src/server/auth-sessions.ts:192]:
```suggestion
```

**Review by flybayer — APPROVED** [review]

**Pull request merged**